src/views/Settings.vue

```diff
@@ -349,44 +355,46 @@ export default {
       } catch (e) {
         this.edgeDeviceError = 'Unavailable. Outdated device?'
       }
-    }
-  },
-  computed: {
-    peerConnectionError: function () {
-      console.log('this.$store.state.pnp.peerConnectionStatus', this.$store.state.pnp.peerConnectionStatus)
-      return this.$store.state.pnp.peerConnectionStatus === PEER_CONNECTION_ERROR
     },
-    ...mapState({
-      peerConnectionStatus: state => state.pnp.peerConnectionStatus,
-      isEdgeConnected: state =>
-        state.pnp.peerConnectionStatus === PEER_CONNECTED,
-      pnp: state => state.pnp,
-      edgePeerId: state => state.pnp.remotePeerId,
-      peerFetch: state => state.pnp.peerFetch,
-      edgeVersion: state => state.edgeDevice.edgeSoftwareVersion
-    }),
-    connectStep: function () {
-      let step = 1
+    handleConnectionStep () {
       switch (this.peerConnectionStatus) {
         case PEER_DISCONNECTED:
+          this.peerConnectionIcon = 'cloud-off-outline'
+          this.peerConnectionText = PEER_DISCONNECTED_TEXT
+          break
         case PEER_CONNECTION_ERROR:
-          step = 1
+          this.connectStep = 1
           break
         case PEER_DISCOVERING:
         case PEER_DISCOVERED:
         case PEER_CONNECTING:
         case PEER_AUTHENTICATING:
-          step = 2
+          this.peerConnectionText = PEER_CONNECTING_TEXT
+          this.connectStep = 2
           break
         case PEER_CONNECTED:
-          step = 3
+          this.connectStep = 3
           break
         default:
           break
       }
-      return step
     }
   },
+  computed: {
+    peerConnectionError: function () {
+      console.log('this.$store.state.pnp.peerConnectionStatus', this.$store.state.pnp.peerConnectionStatus)
+      return this.$store.state.pnp.peerConnectionStatus === PEER_CONNECTION_ERROR
+    },
+    ...mapState({
+      peerConnectionStatus: state => state.pnp.peerConnectionStatus,
+      isEdgeConnected: state =>
+        state.pnp.peerConnectionStatus === PEER_CONNECTED,
+      pnp: state => state.pnp,
+      edgePeerId: state => state.pnp.remotePeerId,
+      peerFetch: state => state.pnp.peerFetch,
+      edgeVersion: state => state.edgeDevice.edgeSoftwareVersion
+    })
+  },
   watch: {
     edgeAddress (value) {
```

Comment:
@vickywane Does this one make sense to you? Why would we update the edgeVersion when the remoteEdgeAddress field changes? Instead, we should update the edgeDetails info AFTER a new peer connection is established.

Reply:
Can you please explain more on this. 

I seem to be lost by this comment. 
The `remoteEdgeAddress` field is not being used here neither is there a new change within this PR that has to do with it.

Reply:
OK, I looked at this again and I don't see an issue with the code. Somehow I confused myself. Thanks for renaming the variable name so it's more obvious what its used for.